src/bio/bio_wal.c

```diff
@@ -1645,6 +1659,24 @@ bio_wal_replay(struct bio_meta_context *mc,
 		D_DEBUG(DB_IO, "Replayed %u WAL transactions\n", nr_replayed);
 		D_ASSERT(wal_id_cmp(si, tx_id, si->si_commit_id) > 0);
 		si->si_unused_id = wal_next_id(si, si->si_commit_id, si->si_commit_blks);
+
+		/* create and set replay metrics */
+		clock_gettime(CLOCK_REALTIME, &tms);
+		e_us = (tms.tv_sec * 1000000) + (tms.tv_nsec / 1000);
+		uuid_unparse_lower(mc->mc_wal->bic_pool_id, pool);
+		rc = d_tm_add_metric(&wal_sz, D_TM_GAUGE, "WAL replay size", "bytes",
+				     "pool/%s/tgt_%d/wal_replay/size", pool, trg);
+		if (rc)
+			D_WARN("Failed to create WAL size replay telemetry: "DF_RC"\n", DP_RC(rc));
+		rc = d_tm_add_metric(&wal_et, D_TM_GAUGE, "WAL replay time", "us",
+				     "pool/%s/tgt_%d/wal_replay/time", pool, trg);
+		if (rc)
+			D_WARN("Failed to create WAL replay ET telemetry: "DF_RC"\n", DP_RC(rc));
+		if (wal_sz)
+			d_tm_set_gauge(wal_sz, total_blk * blk_bytes);
+		if (wal_et)
+			d_tm_set_gauge(wal_et, e_us - s_us);
```

Comment:
besides of "Total replay time" and "Total replayed blocks", I think "Total replayed transactions" might be useful as well? Please use D_TM_DURATION for replayed time and D_TM_STATS_GAUGE for the other metrics.

Reply:
We can certainly add transaction and blocks metrics, but the statistical gauge is probably not needed for the same reason as the comment above - there's only one value.